### PR TITLE
Lang support

### PR DESCRIPTION
**Pull request by AhmedBeheiri**
added language override feature mentioned here https://www.intercom.com/help/en/articles/180-localize-intercom-to-work-with-multiple-languages to support multiple languages based on app locale

### CODE REVIEW

File: example/.flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"intercom_flutter","path":"C:\\\\Users\\\\Sales1\\\\StudioProjects\\\\intercom_flutter\\\\","dependencies":[]}],"android":[{"name":"intercom_flutter","path":"C:\\\\Users\\\\Sales1\\\\StudioProjects\\\\intercom_flutter\\\\","dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"intercom_flutter","dependencies":[]}],"date_created":"2021-02-22 12:28:07.289041","version":"1.22.6"}
```
Comment (Zazo032) [inline example/.flutter-plugins-dependencies]: This file shouldn't be checked into version control

File: example/ios/Flutter/flutter_export_environment.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
```
Comment (Zazo032) [inline example/ios/Flutter/flutter_export_environment.sh]: This file shouldn't be checked into version control

File: lib/intercom_flutter.dart [hunk 3]
```diff
@@ -29,20 +29,24 @@ class Intercom {
   }
 
   static Future<dynamic> setUserHash(String userHash) {
-    return _channel.invokeMethod('setUserHash', {'userHash': userHash});
+    return _channel
+        .invokeMethod('setUserHash', {'userHash': userHash});
```
Comment (Zazo032) [inline lib/intercom_flutter.dart]: Could you revert format changes from this file?

**Review by Zazo032 — CHANGES_REQUESTED** [review]
Thanks @AhmedBeheiri for the PR, this will be a nice addition :)

I left some comments, please have a look.

Also, there seem to be a few conflicts that need to be resolved.

Could you also add or modify an existing test to ensure the `lang` param is being passed properly?

File: test/intercom_flutter_test.dart [hunk 4]
```diff
@@ -154,6 +154,7 @@ void main() {
         phone: '+37256123456',
         company: 'Some Company LLC',
         companyId: '2',
+        lang: 'en'
```
Comment (Zazo032) [inline test/intercom_flutter_test.dart]:
```suggestion
        lang: 'en',
```

**Review by Zazo032 — APPROVED** [review]
Thanks! Only missing thing now is adding your changes to `CHANGELOG.md` and bump the plugin version on `pubspec.yaml` (2.4.0).

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+##2.4.0
+* added language override feature to change language based on app locale
```
Comment (Zazo032) [inline CHANGELOG.md]:
```suggestion
* Added language override feature to `updateUser` method.
```

File: lib/intercom_flutter.dart [hunk 6]
```diff
@@ -79,7 +84,8 @@ class Intercom {
     return _channel.invokeMethod('logout');
   }
 
-  static Future<dynamic> setLauncherVisibility(IntercomVisibility visibility) {
+  static Future<dynamic> setLauncherVisibility(
+      IntercomVisibility visibility) {
```
Comment (Zazo032) [inline lib/intercom_flutter.dart]: Revert formatting change here

File: lib/intercom_flutter.dart [hunk 7]
```diff
@@ -114,13 +120,14 @@ class Intercom {
...
-        .invokeMethod('logEvent', {'name': name, 'metaData': metaData});
+    return _channel.invokeMethod(
+        'logEvent', {'name': name, 'metaData': metaData});
   }
 
   static Future<dynamic> sendTokenToIntercom(String token) {
     print("Start sending token to Intercom");
-    return _channel.invokeMethod('sendTokenToIntercom', {'token': token});
+    return _channel
+        .invokeMethod('sendTokenToIntercom', {'token': token});
```
Comment (Zazo032) [inline lib/intercom_flutter.dart]: Revert formatting change here

File: lib/intercom_flutter.dart [hunk 8]
```diff
@@ -132,7 +139,8 @@ class Intercom {
         .invokeMethod('displayMessageComposer', {'message': message});
   }
 
-  static Future<bool> isIntercomPush(Map<String, dynamic> message) async {
+  static Future<bool> isIntercomPush(
+      Map<String, dynamic> message) async {
```
Comment (Zazo032) [inline lib/intercom_flutter.dart]: Revert formatting change here

**Review by Zazo032 — CHANGES_REQUESTED** [review]
There are still a few conflicts that need to be addressed, I can't merge this PR until they're fixed.

<details>
<summary>Conflicting files</summary>

```
CHANGELOG.md
android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt
ios/Classes/IntercomFlutterPlugin.m
lib/intercom_flutter.dart
pubspec.yaml
test/intercom_flutter_test.dart
```
</details>

Also, I left some minor nits.

**Comment by Zazo032** [comment]
There are still conflicts:

image

**Comment by AhmedBeheiri** [comment]
> There are still conflicts:
> 
> image

please check again now

**Comment by Zazo032** [comment]
I still see conflicts with your branch. If you can't fix them, you may want to close this PR, create a new branch from `master`, push your changes again, and then create a new PR

**Pull request closed without merging**